src/hivpy/hiv_testing.py

```diff
@@ -50,81 +55,125 @@ def __init__(self, **kwargs):
         self.covid_disrup_affected = False
         self.testing_disrup_covid = False
 
-    def update_hiv_testing(self, pop):
+    def update_hiv_testing(self, pop, time_step: timedelta):
         """
         Update which individuals in the population have been tested.
         COVID disruption is factored in.
         """
-        # testing occurs after a certain year if there is no covid disruption
-        if ((pop.date.year >= self.date_start_testing)
-           & (not (self.covid_disrup_affected | self.testing_disrup_covid))):
+        # mark people for testing
+        # hiv symptomatic > non hiv symptomatic > vmmc > anc > self testing > general > prep
+        self.test_mark_hiv_symptomatic(pop)
+        self.test_mark_non_hiv_symptomatic(pop)
+        self.test_mark_vmmc(pop, time_step)
+        self.test_mark_anc(pop, time_step)
+        self.test_mark_general_pop(pop)
```

Comment:
In the SAS code the order appears to be:
1. Test for non HIV TB (doesn't appear to exist in HIVpy yet)
2. Circumcision related testing
3. Non HIV symptoms
4. ANC
5. Post Delivery
6. General population (including sex workers)
7. Testing before prep (not in HIVpy yet)
8. routine testing on prep (not in HIVpy yet)
9. HIV symptomatic testing

Reply:
I'm not sure how the discussion of VMMC vs HIV ordering affects this precisely, but it does seem like the HIV symptomatic needs tobe moved down to the bottom

Reply:
Push ANC / post delivery above non-hiv symptoms